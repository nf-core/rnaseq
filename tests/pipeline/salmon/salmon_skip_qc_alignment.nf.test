nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    options "-profile test,docker"
    tag "pipeline"

    test("SALMON (Skip QC & Alignment) (GSE110004)") {

        when {
            params {
                outdir         = "$outputDir"
                pseudo_aligner = "salmon"
                skip_qc        = true
                skip_alignment = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/pipeline_info/software_versions.yml")).match("software_versions") },
                { assert snapshot(path("$outputDir/pipeline_info/samplesheet.valid.csv")).match("samplesheet") },
                { assert snapshot(path("$outputDir/bbsplit/RAP1_IAA_30M_REP1.stats.txt"),
                                path("$outputDir/bbsplit/RAP1_UNINDUCED_REP1.stats.txt"),
                                path("$outputDir/bbsplit/RAP1_UNINDUCED_REP2.stats.txt"),
                                path("$outputDir/bbsplit/WT_REP1.stats.txt")).match("bbsplit") },
                { assert new File("$outputDir/bbsplit/WT_REP2.stats.txt").exists() },
                { assert snapshot(path("$outputDir/multiqc/multiqc_data/multiqc_cutadapt.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_fastqc.txt")).match("multiqc") },
                { assert new File("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt").exists() },
                { assert snapshot(path("$outputDir/salmon/salmon_tx2gene.tsv")).match("salmon") },
                { assert new File("$outputDir/salmon/salmon.merged.gene_counts.tsv").exists() },
                { assert new File("$outputDir/salmon/salmon.merged.transcript_counts.tsv").exists() },
                { assert new File("$outputDir/salmon/RAP1_IAA_30M_REP1/quant.sf").exists() },
                { assert new File("$outputDir/salmon/RAP1_IAA_30M_REP1/quant.genes.sf").exists() },
                { assert new File("$outputDir/salmon/RAP1_UNINDUCED_REP1/quant.sf").exists() },
                { assert new File("$outputDir/salmon/RAP1_UNINDUCED_REP1/quant.genes.sf").exists() },
                { assert new File("$outputDir/salmon/RAP1_UNINDUCED_REP2/quant.sf").exists() },
                { assert new File("$outputDir/salmon/RAP1_UNINDUCED_REP2/quant.genes.sf").exists() },
                { assert new File("$outputDir/salmon/WT_REP1/quant.sf").exists() },
                { assert new File("$outputDir/salmon/WT_REP1/quant.genes.sf").exists() },
                { assert new File("$outputDir/salmon/WT_REP2/quant.sf").exists() },
                { assert new File("$outputDir/salmon/WT_REP2/quant.genes.sf").exists() },
                { assert new File("$outputDir/trimgalore/RAP1_IAA_30M_REP1_1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/RAP1_IAA_30M_REP1_2.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/RAP1_UNINDUCED_REP1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/RAP1_UNINDUCED_REP2.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/WT_REP1_1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/WT_REP1_2.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/WT_REP2_1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/WT_REP2_2.fastq.gz_trimming_report.txt").exists() }
            )
        }

    }

}
