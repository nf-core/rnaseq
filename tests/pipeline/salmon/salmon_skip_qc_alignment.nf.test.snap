{
    "samplesheet": {
        "content": [
            "samplesheet.valid.csv:md5,3d40bf3d424290cec0dbe9ee95b7dd12"
        ],
        "timestamp": "2023-05-08T18:09:32+0000"
    },
    "software_versions": {
        "content": [
            "software_versions.yml:md5,017fc93865d2c889c721268e6e0f6e32"
        ],
        "timestamp": "2023-05-08T18:09:32+0000"
    },
    "salmon": {
        "content": [
            "salmon_tx2gene.tsv:md5,b367acfc4b7df7fb68fb1b1d2956e27e"
        ],
        "timestamp": "2023-05-08T18:09:32+0000"
    },
    "bbsplit": {
        "content": [
            "RAP1_IAA_30M_REP1.stats.txt:md5,a87292c6f7166096166a9015c2e63dac",
            "RAP1_UNINDUCED_REP1.stats.txt:md5,781e49024150a8313779c574b58ad035",
            "RAP1_UNINDUCED_REP2.stats.txt:md5,e455d0163bf2d988224b0c0092605213",
            "WT_REP1.stats.txt:md5,6eab6936804c1157d5e1c61c0562d7de"
        ],
        "timestamp": "2023-05-08T18:09:32+0000"
    },
    "multiqc": {
        "content": [
            "multiqc_cutadapt.txt:md5,aac9581a5670cb55edf564f3d6c1f9a7",
            "multiqc_fastqc.txt:md5,e3ebd5c2676bebb3c0176c4e985f3580"
        ],
        "timestamp": "2023-05-08T18:09:32+0000"
    }
}